tests/testthat/test-get_foi_index.R

```diff
@@ -0,0 +1,86 @@
+# Setup for testing ----
+# Sample survey features to use in test
+survey_features <- data.frame(
+  age_min = c(1, 6, 11, 16, 21),
+  age_max = c(5, 10, 15, 20, 25),
+  survey_year = 2025
+)
+
+# Test get_foi_index ----
+test_that("get_foi_index returns correct output for valid model types", {
+  # Test for model_type "age"
+  result_age <- get_foi_index(survey_features, group_size = 5, model_type = "age")
+
+  # Check if the data frame has the correct structure for "age"
+  checkmate::assert_names(
+    names(result_age),
+    must.include = c("age", "foi_index")
+  )
+  expect_equal(nrow(result_age), max(survey_features$age_max))
```

Comment:
I think there are more things we could check: e.g. max index, all indices represented up until that max index, that we only have contiguous indices. I realise this may seem overkill given the simplicity of this function, but we could imagine changing this function in the future and we'd want to check that it still works as intended.

Reply:
I added validations to check whether the indexes are consecutive, meaning that we are not accepting any of the cases that you mentioned above (I think). I'm doing so by checking that `diff(foi_index$foi_index)` only contains either 0s or 1s. Any negative difference would correspond to non-consecutive indexes like 1,1,2,2,1,..., whereas any difference higher than 1 would correspond to something like 1,1,3,3,... I think this cover at least the most urgent exceptions.